enterprise/frontend/src/metabase-enterprise/data-studio/data-model/components/TablePicker/types.ts

```diff
@@ -133,3 +141,7 @@ export function isTableNode(
       node.children.every((child) => child.type === "table"))
   );
 }
+
+export function isTableNode(node: FlatItem): node is ExpandedTableItem {
```

Comment:
nit: this helper could be used in many existing places.

`isDatabaseNode` probably as well.